great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,213 @@
+        candidate_regexes: Union[
+            Set[str],
+            List[str],
+            "RegexPatternStringParameterBuilder.CANDIDATE_REGEX",  # noqa: F821
+        ] = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=self.candidate_regexes,
+            expected_return_type=None,
+            variables=variables,
+            parameters=parameters,
+        )
+        if candidate_regexes is not None and isinstance(candidate_regexes, list):
+            candidate_regexes = set(candidate_regexes)
+        else:
+            candidate_regexes = RegexPatternStringParameterBuilder.CANDIDATE_REGEX
+
+        regex_string: str
+        match_regex_metric_value_kwargs: dict
+        for regex_string in candidate_regexes:
+            if self.metric_value_kwargs:
+                match_regex_metric_value_kwargs: dict = {
+                    **self._metric_value_kwargs,
+                    **{"regex": regex_string},
+                }
+            else:
+                match_regex_metric_value_kwargs: dict = {"regex": regex_string}
+
+            metric_computation_result: MetricComputationResult = self.get_metrics(
+                metric_name="column_values.match_regex.unexpected_count",
+                metric_domain_kwargs=self.metric_domain_kwargs,
+                metric_value_kwargs=match_regex_metric_value_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            )
+            metric_values = metric_computation_result.metric_values
+            # Now obtain 1-dimensional vector of values of computed metric (each element corresponds to a Batch ID).
+
+            metric_values = metric_values[:, 0]
+            match_regex_unexpected_count: int = sum(metric_values)
+            success_ratio: float = (
+                nonnull_count - match_regex_unexpected_count
+            ) / nonnull_count
+            regex_string_success_ratios[regex_string] = success_ratio
+
+        best_regex_string: Optional[str] = None
+        best_ratio: float = 0.0
+        # Obtain threshold from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        threshold: float = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=self._threshold,
+            expected_return_type=float,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        regex_string: str
+        ratio: float
+        for regex_string, ratio in regex_string_success_ratios.items():
+            if ratio > best_ratio and ratio >= threshold:
+                best_regex_string = regex_string
+                best_ratio = ratio
+
+        parameter_values: Dict[str, Any] = {
+            f"$parameter.{self.name}": {
+                "value": best_regex_string,
+                "details": {"success_ratio": best_ratio},
```

Comment:
1. For `value` instead of the single best, let's include all of the ones that are above the target threshold, sorted by ratio.
2. Let's include the success ratios for all of the regexes that we tried.

Reply:
@jcampbell to clarify : would 2 include all regexes above the threshold? Or the full list? (I'm understanding 1 and 2 to be describing slightly different things)

Reply:
updated PR to include all REGEX values and success ratios that pass threshold. Added a test that tests multiple matches too